### ISSUE TITLE
Tokens with pictures / alpha-block placeholders and names shown when composing messages should be optional

### ISSUE THREAD

**Issue by njeyaakili** [k9mail/k-9#1207]
### Expected behaviour

general/display correspondent display settings -- email address and pictures -- should be respected.

1 - when composing a message, correspondent email address should be displayed (when the general/display setting for showing correspondent name is unchecked), not their name (from the addressbook)

2 - the correspondent picture (or alpha block placeholder) should not be displayed if the general/display setting "show contact picture" is unchecked.
### Actual behaviour

when 
if correspondent is in addressbook, the addressbook name is displayed.

the correspondent's picture (or if one doesn't exist, a placeholder alpha block) is displayed
### Steps to reproduce

1.originate a message
2.enter a correspondent's email address
3.if in addressbook, contact name and picture (or alpha block if no picture or not in addressbook) will be displayed.
### Environment

K-9 Mail version:5.108

Android version:6.0.1 (but this doesn't matter)

Account type (IMAP, POP3, WebDAV/Exchange):imap 

in general/display settings, "show correspondent names, rather than email addresses" and "show contact pictures" are both unchecked.

**Comment by philipwhiuk** [comment]
Have you tried exiting and re-entering the app after changing those settings?

**Comment by njeyaakili** [comment]
Yes. I have always had these options turned off (unchecked), but with the upgrade from .107 to .108 these checkbox preferences don't seem to be being respected when composing a message. They work as expected when displaying a message list.

For additional testing I:
--  turned these options on, that got me pictures &etc. on the message list, as well as with compose, as expected.
-- turned them off (unchecked), the message list returned to addresses/no pictures, compose still showed pictures.
-- did a "force close" in the application manager
-- reopened .108,  the compose still displayed the name and the picture (or placeholder alpha) on correspondent addresses that I entered (as well as on my default "copy to" bcc, which is now (new with .108 I believe) displayed).

**Comment by philipwhiuk** [comment]
Oh when you say picture you mean the token things in To/Cc on composition?

I don't know whether that was ever intended to be controlled / related to the contacts/pictures setting. You need some way of marking email addresses as valid etc.

Contacts and pictures on composition might be a separate setting - the idea behind enabling / disabling the other two is about space on the message list.

**Comment by njeyaakili** [comment]
Yes, the picture/contact name (rather than email address) on composition.

The contacts and pictures on composition are new (for me, with my settings, at least) with .108.  If they can't be/aren't controlled by the existing contacts/pictures settings then they need their own setting(s). 

As the display of contacts/pictures on composition is new, I did look under all the options that I could think of as relevant to see if there was a new setting -- e.g., account/sending mail/composition defaults, general/display, etc. and found no new settings. 

I'm after the ability to have the clean message display, including the ability to clearly see the correspondent address being used (and without the unnecessary picture), that has been available up to now.

**Comment by njeyaakili** [comment]
I have attached compose screens from .107 and .108 to this comment to make this clearer. The K9 settings are unaltered between these two (simply updated from .107 to .108).  [the global/display/show correspondent names and show contact pictures message-list options are both unchecked.]

With .108 it is acting as if the message-list settings global/display/show correspondent names and show contact pictures are being applied. The previous discussion indicates that those are message-list specific, so shouldn't/don't apply to the composition screen (which I can confirm is the case with .107). So, either those settings need to also apply to the compose screen or there need to be additional compose screen (either at the global or account level) options that can make it so that  a) the correspondent's address, rather than contacts entry name is displayed and b) no contact picture/token is displayed.

Additionally, with .108 the "bcc all messages to" address is displayed even though the cc/bcc addresses weren't expanded. Obviously that address is displayed when the cc/bcc are expanded (and that's what happens in .107), but I don't see a reason for it to be displayed by default.

![compose-screen-5 107](https://cloud.githubusercontent.com/assets/17148103/14061514/a9dc8086-f379-11e5-91cb-a65f8664520a.png)
![compose-screen-5 108](https://cloud.githubusercontent.com/assets/17148103/14061517/b17d69e0-f379-11e5-8327-d6ad778281c4.png)

**Comment by njeyaakili** [comment]
Is there an intent to change the (new - with .108) behavior on how addresses are displayed on compose (optionally) back to what was in place with .107? My expectation as a user would be that if I make the selection to not show correspondent pictures (or the block letter placeholders) when viewing a received message, that the same would apply to addressing on messages when composing. So that an additional setting is not required, I would think that the existing "global/display/show contact pictures" setting could be made to apply to how "contact pictures" are handled in compose in addition to its current application on message display.

Part of this desire is get rid of the unnecessary "clutter" that the pictures and placeholders add. The additional issue is clarity. When composing I want to be able to clearly see the correspondent's email address at any point during the compose, regardless of its source and without having to take additional actions. Currently, the actual email address of something from "contacts" is only visible during the initial selection. After that it's obscured by the contact "name" and a tag, which has to be selected to see the actual contact entry.

It would also be nice if the bcc wasn't automatically expanded. It doesn't seem necessary to always show the "bcc all messages to" address (and this wasn't shown in .107). Instead, give a way to expand that field so someone can add bcc's (in addition to seeing what's there).

**Comment by Valodim** [comment]
This is low prio for me, but if someone wants to work on it, PRs welcome.

**Comment by philipwhiuk** [comment]
I'm not big on us having yet another option or maintaining an old style of layout screen, for the same reason that we won't maintain old style interfaces if we adopt Material. Upgrading will change the UI and drowning K-9 in settings to accommodate showing all previous versions of the UI isn't realistic.

If it's reasonable to put it an existing option fine.

**Comment by njeyaakili** [comment]
I don't think that there's a need for an additional option.

Rather, I'm all for making these settings for compose part of the existing "[global] settings/display/show contact names" and ".../show contact pictures" options that apply to message display. That way the user-selected interface settings would be consistent for both message display and message compose. As it is now, the user can control how these items are presented in message display but not message  compose.

**Comment by dllud** [comment]
@njeyaakili is right on spot here. Having that setting control the way contacts are displayed in message display, as well as message composition, would do the trick.

The move to these new contact tokens seems to have been too hasty. They do look good and seem a good way to polish K-9's UI. However, they way they are implemented right now broke a lot of functionality, as you can see from the several issues related to them that have been opened in the past few weeks. These tokens need some more work to become usable, and thus, IMHO they shouldn't be the default option yet, or at least there should be a way to disable them.

**Comment by philipwhiuk** [comment]
> However, they way they are implemented right now broke a lot of functionality, as you can see from the several issues related to them that have been opened in the past few weeks.

That's not because we've been hasty. The release was in unstable for months and months and for most of that time the tokens were present. Nobody raised any issues or we'd have held back from the release.

The problem is lack of beta testers. 

Making them optional sounds nice, but as soon as you removed that option everyone would moan almost as much as they are now. And leaving more options in, as I have said, is not practical.

**Comment by njeyaakili** [comment]
I opened this item in March, I believe as soon as these placeholders first showed up in the unstable/alpha release. In addition to the UI inconsistency between the message display and compose I also pointed out that it was almost impossible to see the email address actually being used. I honestly stopped using the alpha versions on my primary device because I found these placeholders to be so annoying and intrusive in the limited real estate available on a mobile device screen, so didn't run into some of the other issues that have been reported of late.

I'm fine with having things (placeholders, etc.) like this, for people who what them, but given the size of a mobile device display there ideally needs to be a way for the user (who wants content/context over glitz) to get a clean, minimalist, display. One option, that applies globally, would do it - e.g., by including the display of the placeholders, etc., in message composition in with the existing message display "[global] settings/display/show contact names" and ".../show contact pictures" options. Gmail, for instance, has display settings of "comfortable", "cozy" and "compact". They seem to only address line spacing, but the same concept of "minimalist" and "pretty" (or whatever it a reasonable label, e.g., with placeholders/pictures and names vs. email addresses) might be applicable here.

**Comment by dllud** [comment]
> The problem is lack of beta testers.

I see that your beta channel is at Google Play and here on GitHub Releases. Have you considered adding beta versions to F-Droid Archive? Most F-Droid users are tech savvy, I think you could easily find new beta testers there.

> And leaving more options in, as I have said, is not practical.

As @njeyaakili points out, right now there is no need for an extra option. It is just a matter of applying the existing option to both message display and message composition.

In the future I would be ok with tokens only, provided that issues like this one (showing email addresses) and others like #1991 got fixed.

**Comment by philipwhiuk** [comment]
@cketti - you had comments on this in another issue. What do you think about moving the option about contact pictures to elsewhere in the settings and making it apply to this?

Regarding F-Droid, the release process is fairly slow and there's no way for users to sign up to a beta channel AFAIK. I think that's why we haven't been beta testing there.

In terms of what is displayed (names vs email addresses), that shouldn't be too hard to fix so I'll look at that   and #1991

**Comment by dllud** [comment]
> Regarding F-Droid, [...] there's no way for users to sign up to a beta channel AFAIK.

There is no explicit way, that's why I mentioned [F-Droid Archive](https://f-droid.org/wiki/page/Known_Repositories) and not the main repo. What I've seen others do is to add a new app but just to the Archive repo, something like "K-9 Mail Beta", and make the beta releases there. For instance, that is what F-Droid devs are currently doing with their build of Firefox Mobile, dubbed Fennec F-Droid.
Another option is to setup your own repo and release beta versions there.
